sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/write/WriteDocumentsService.java

```diff
@@ -459,12 +460,24 @@ private Uni<Integer> determineTtl(
         .getDocumentTtlInfo(namespace, collection, documentId, ctx)
         .onItem()
         .transform(
-            rawDocument ->
-                rawDocument
-                    .rows()
-                    .get(0)
-                    .getLong(String.format("ttl(%s)", configuration.table().leafColumnName()))
-                    .intValue())
+            rawDocument -> {
+              if (rawDocument == null) {
+                return 0;
+              }
+              List<RowWrapper> rows = rawDocument.rows();
+              if (rows.isEmpty()) {
+                return 0;
+              }
+              try {
+                Long value =
+                    rows.get(0)
+                        .getLong(String.format("ttl(%s)", configuration.table().leafColumnName()));
+                return value.intValue();
+              } catch (IllegalArgumentException e) {
+                // getLong had a NULL value
+                return 0;
+              }
```

Comment:
check that the `columnExists` to avoid throwing this exception all the time, as this is on the valid write path (exceptions are not free):

```
RowWrapper wrapper = rows.get(0);
String column ="ttl(%s)".formatted(configuration.table().leafColumnName());
if (wrapper.columnExists(column) {
    return wrapper.getLong(column);
} else {
   return 0;
}
```

Reply:
The problem here is that the column exists, but getting its value is returning a null value when a non-null number value is expected (that is the IllegalArgumentException I am intercepting). I tried to change it to what you wrote above but am still encountering the same issue; an unexpected error occurs reading `Expected INT value, received NULL`

In order to fix this, I would probably have to change the way values are accessed in RowWrapper, or maybe create a "getOptionaLong" method instead...

Reply:
Wait is there any way we can add the `wrapper.isNull()` call? @olim7t quick help?

Reply:
@EricBorczuk could be this:

  boolean isNull(String columnName) {
    int i = firstIndexOf(columnName);
    if (i < 0) {
      throw new IllegalArgumentException(String.format("Column '%s' does not exist", columnName));
    }
     Value value = row().getValues(i);
     return value.hasNull();
  }

Reply:
Yep this would work, I'll add it

Reply:
Forget it, there is already `isNull` method!